File: app/src/main/java/com/example/andrasnemeth/nnotification/MainActivity.java
package com.example.andrasnemeth.nnotification;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity  extends AppCompatActivity {

    private LegacyNotifications legacyNotifications;
    private InlineReply inlineReply;
    private BundledNotifictions bundledNotifictions;
    private CustomViewNotifications customViewNotifications;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BaseNotificationHelper notificationHelper = new BaseNotificationHelper(this);
        legacyNotifications = new LegacyNotifications(notificationHelper);
        inlineReply = new InlineReply(notificationHelper);
        bundledNotifictions = new BundledNotifictions(notificationHelper);
        customViewNotifications = new CustomViewNotifications(notificationHelper);

        inlineReply.handleInlineReply(getIntent());
    }

    public void onPlainButtonClick(View view) {
        legacyNotifications.showPlainNotification();
    }

    public void onInboxStyleButtonClick(View view) {
        legacyNotifications.showInboxSytleNotification();
    }

    public void onBigPictureStyleButtonClick(View view) {
        legacyNotifications.showBigPictureStyleNotification();
    }

    public void onBigTextStyleButtonClick(View view) {
        legacyNotifications.showBigTextStyleNotification();
    }

    public void onInlineReplyButtonClick(View view) {
        inlineReply.showNotification();
    }

    public void onBundledButtonClick(View view) {
        bundledNotifictions.addNotificationToBundle();
    }

    public void onCustomViewButtonClick(View view) {
        customViewNotifications.show();
    }
}

File: app/src/main/java/com/example/andrasnemeth/nnotification/MainActivity.kt
package com.example.andrasnemeth.nnotification

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var legacyNotifications: LegacyNotifications? = null
    private var inlineReply: InlineReply? = null
    private var bundledNotifictions: BundledNotifictions? = null
    private var customViewNotifications: CustomViewNotifications? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val notificationHelper = BaseNotificationHelper(this)
        notificationHelper.createNotificationChannel()
        legacyNotifications = LegacyNotifications(notificationHelper)
        inlineReply = InlineReply(notificationHelper)
        bundledNotifictions = BundledNotifictions(notificationHelper)
        customViewNotifications = CustomViewNotifications(notificationHelper)
        inlineReply!!.handleInlineReply(intent)
    }

    fun onPlainButtonClick(view: View?) {
        legacyNotifications!!.showPlainNotification()
    }

    fun onInboxStyleButtonClick(view: View?) {
        legacyNotifications!!.showInboxSytleNotification()
    }

    fun onBigPictureStyleButtonClick(view: View?) {
        legacyNotifications!!.showBigPictureStyleNotification()
    }

    fun onBigTextStyleButtonClick(view: View?) {
        legacyNotifications!!.showBigTextStyleNotification()
    }

    fun onInlineReplyButtonClick(view: View?) {
        inlineReply!!.showNotification()
    }

    fun onBundledButtonClick(view: View?) {
        bundledNotifictions!!.addNotificationToBundle()
    }

    fun onCustomViewButtonClick(view: View?) {
        customViewNotifications!!.show()
    }
}